src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,313 @@
+            this.sampleInfos = sampleInfos;
+        }
+
+        @Override
+        public void visit(Event event, long samples, long value) {
+            double factor = counterFactor();
+            value = factor == 1.0 ? value : (long) (value * factor);
+
+            long msFromStart = (event.time - jfr.chunkStartTicks) * 1_000 / jfr.ticksPerSec;
+            long timeNanos = jfr.chunkStartNanos + msFromStart * 1_000_000;
+
+            Range range = idToRange.computeIfAbsent(event.stackTraceId, this::computeLocationRange);
+            sampleInfos.add(
+                    new SampleInfo(samples, value, range, getThreadName(event.tid), timeNanos));
+        }
+
+        // Range of values in Profile.location_indices
+        private Range computeLocationRange(int stackTraceId) {
+            StackTrace stackTrace = jfr.stackTraces.get(stackTraceId);
+            if (stackTrace == null) {
+                return new Range(0, 0);
+            }
+            for (int i = 0; i < stackTrace.methods.length; ++i) {
+                Line line = makeLine(stackTrace, i);
+                int locationIdx = linePool.index(line);
+                otlpProto.writeInt(locationIdx);
+            }
+            nextLocationIdx += stackTrace.methods.length;
+            return new Range(nextLocationIdx, stackTrace.methods.length);
+        }
+    }
+
+    private Line makeLine(StackTrace stackTrace, int i) {
+        String methodName = getMethodName(stackTrace.methods[i], stackTrace.types[i]);
+        int lineNumber = stackTrace.locations[i] >>> 16;
+        int functionIdx = functionPool.index(methodName);
+        return new Line(functionIdx, lineNumber);
+    }
+
+    private static final class SampleInfo {
+        final long samples;
+        final long value;
+        // Indices into ProfilesDictionary.location_table
+        final Range locationsRange;
+        final String threadName;
+        final long timeNanos;
+
+        public SampleInfo(
+                long samples, long value, Range locationsRange, String threadName, long timeNanos) {
+            this.samples = samples;
+            this.value = value;
+            this.locationsRange = locationsRange;
+            this.threadName = threadName;
+            this.timeNanos = timeNanos;
+        }
+    }
+
+    private static final class Range {
+        final long start;
+        final long length;
+
+        public Range(long start, long length) {
+            this.start = start;
+            this.length = length;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (!(o instanceof Range)) return false;
+
+            Range range = (Range) o;
+            return start == range.start && length == range.length;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            result = (int) (31 * result + start);
+            return (int) (31 * result + length);
+        }
+    }
```

Comment:
If a class is not used as a key in collections, these methods are not needed, they just add noise.

Reply:
Right, this is a leftover from a previous iteration. Fixed in 2359e43708a736b01714e3842adfd6a91ad6e203